homeassistant/components/caldav/calendar.py

```diff
@@ -199,21 +199,47 @@ def update(self):
             dt.start_of_local_day(), dt.start_of_local_day() + timedelta(days=1)
         )
 
+        # Create new vevents for each recurrence of an event that happens today.
+        # For recurring events, some servers return the original event with recurrence rules
+        # and they would not be properly parsed using their original start/end dates.
+        vevents = [event.instance.vevent for event in results]
+        new_vevents = []
+        for vevent in vevents:
+            rrules = vevent.getrruleset()
+            if rrules:
+                occurrences = []
+                for rrule in rrules:
+                    if not rrule.tzinfo:
+                        rrule = dt.as_local(rrule)
+                    if (
+                        dt.start_of_local_day() + timedelta(days=1)
```

Comment:
let's calculate these outside of the loop.

Reply:
How?

Reply:
put outside of loop:
```python
today_start = dt.start_of_local_day()
tomorrow = today_start + timedelta(days=1)
```

Reply:
Oh, sure. I thought you were referring to line 212. :)

Reply:
I need to learn how to read GitHub comments.